src/snapred/backend/service/CalibrationService.py

```diff
@@ -344,10 +346,24 @@ def _collectMetrics(self, focussedData, focusGroup, pixelGroupingParam):
         )
         return FocusGroupMetric(focusGroupName=focusGroup.name, calibrationMetric=metric)
 
+    @FromString
+    def readQuality(self, runId: str, version: str):
+        calibrationRecord = self.dataFactoryService.getCalibrationRecord(runId, version)
+        if calibrationRecord is None:
+            raise ValueError(f"No calibration record found for run {runId}, version {version}.")
+        GenerateCalibrationMetricsWorkspaceRecipe().executeRecipe(
+            CalibrationMetricsWorkspaceIngredients(calibrationRecord=calibrationRecord)
+        )
+        for ws_name in calibrationRecord.workspaceNames:
+            path = self.dataFactoryService.getCalibrationDataPath(
+                calibrationRecord.runNumber, calibrationRecord.version
+            )
+            fullPath = os.path.join(path, ws_name + ".nxs")
+            self.dataFactoryService.loadWorkspace(fullPath, ws_name)
```

Comment:
The part where it finds the path should probably live in the data layer.

I would condense these into a `loadCalibrationDataWorkspace` method.